lib/fog/vsphere/requests/compute/create_vm.rb

```diff
@@ -106,10 +107,18 @@ def device_change attributes
         def boot_options attributes
           # NOTE: you must be using vsphere_rev 5.0 or greater to set boot_order
           # e.g. Fog::Compute.new(provider: "vsphere", vsphere_rev: "5.5", etc)
-          return unless @vsphere_rev.to_f >= 5
-          RbVmomi::VIM::VirtualMachineBootOptions.new(
-            :bootOrder => boot_order(attributes)
-          )
+          options = {}
+          if @vsphere_rev.to_f >= 5 and attributes[:boot_order]
+            options[:bootOrder] = boot_order(attributes)
+          end
+
+          # Set attributes[:boot_retry] to a delay in miliseconds to enable boot retries
+          if attributes[:boot_retry]
+            options[:bootRetryEnabled] = true
```

Comment:
Got it.

But if the api returns those attributes i think it is nice to have them mapped in the model.

Not so long ago i've introduced the option `as` to the `attribute` DSL which is supposed to do the opposite of the option `aliases`.

It will map a simple attribute name like `boot` to something the api expects like `Boot`. But it only works for attribute mapping so far. If you need to transform one attribute into a structure then it does not work (yet).

Do you think it would fit the need here?